exporter/chronicleexporter/marshal.go

```diff
@@ -177,6 +215,35 @@ func (m *protoMarshaler) getLogType(ctx context.Context, logRecord plog.LogRecor
 	return m.cfg.LogType, nil
 }
 
+func (m *protoMarshaler) getNamespace(ctx context.Context, logRecord plog.LogRecord, scope plog.ScopeLogs, resource plog.ResourceLogs) (string, error) {
+	namespace, err := m.getRawField(ctx, chronicleNamespaceField, logRecord, scope, resource)
+	if err != nil {
+		return m.cfg.Namespace, fmt.Errorf("get chronicle log type: %w", err)
+	}
+	if namespace != "" {
+		return namespace, nil
+	}
+	return m.cfg.Namespace, nil
+}
+
+func (m *protoMarshaler) getIngestionLabels(logRecord plog.LogRecord) ([]*api.Label, error) {
+	configLabels := make([]*api.Label, 0)
+	for key, value := range m.cfg.IngestionLabels {
+		configLabels = append(configLabels, &api.Label{
+			Key:   key,
+			Value: value,
+		})
+	}
+	ingestionLabels, err := m.getRawNestedFields(chronicleIngestionLabelsPrefix, logRecord)
+	if err != nil {
+		return configLabels, fmt.Errorf("get chronicle ingestion labels: %w", err)
+	}
+	if len(ingestionLabels) != 0 {
+		return ingestionLabels, nil
+	}
+	return configLabels, nil
+}
```

Comment:
Again, the return value isn't used when there's an error, so return an empty slice and simplify this code a bit
```suggestion
func (m *protoMarshaler) getIngestionLabels(logRecord plog.LogRecord) ([]*api.Label, error) {
	// check for labels in attributes["chronicle_ingestion_labels"]
	ingestionLabels, err := m.getRawNestedFields(chronicleIngestionLabelsPrefix, logRecord)
	if err != nil {
		return []*api.Label{}, fmt.Errorf("get chronicle ingestion labels: %w", err)
	}
	if len(ingestionLabels) != 0 {
		return ingestionLabels, nil
	}

	// use labels defined in config if needed
	configLabels := make([]*api.Label, 0)
	for key, value := range m.cfg.IngestionLabels {
		configLabels = append(configLabels, &api.Label{
			Key:   key,
			Value: value,
		})
	}

	return configLabels, nil
}
```